flexget/plugins/sites/lostfilm.py

```diff
@@ -47,151 +37,112 @@ class LostFilm(object):
+                    lostfilm_num, season_num, episode_num = [int(x) for x in seen_btn['data-code'].split('-')]
+                except AttributeError:
+                    log.exception('Could not parse %s, probably HTML markup was changed', release_url)
                     continue
-                if quality_map.get(quality):
-                    quality, file_ext = quality_map.get(quality)
-                else:
-                    file_ext = 'avi'
-                if series_name_eng:
-                    new_title = '.'.join(
-                        [
-                            series_name_eng,
-                            episode_id,
-                            quality,
-                            'rus.LostFilm.TV',
-                            file_ext,
-                            'torrent'
-                        ]
-                    ).replace(' ', '.')
-                else:
-                    if item.get('title') is not None:
-                        new_title = '{} {}'.format(
-                            item['title'],
-                            quality)
-                    else:
-                        log.debug('Item doesn\'t have a title')
-                        continue
-                new_entry = Entry()
-                new_entry['url'] = torrent_link
-                new_entry['title'] = new_title.strip()
-                if series_name_rus:
+
+                torrents = self.get_torrents(task, lostfilm_num, season_num, episode_num)
+                for quality, torrent_link in torrents.items():
+                    new_entry = Entry()
+                    new_entry['title'] = '{:s} S{:02d}E{:02d} - {:s} [{:s}]'.format(series_name_eng, season_num, episode_num, episode_name_eng, quality)
+                    new_entry['url'] = torrent_link
+                    new_entry['name'] = series_name_eng
+                    new_entry['season'] = season_num
+                    new_entry['episode'] = episode_num
                     new_entry['series_name_rus'] = series_name_rus
-                if episode_name_rus:
                     new_entry['episode_name_rus'] = episode_name_rus
-                if series_name_eng:
                     new_entry['series_name_eng'] = series_name_eng
-                if episode_name_eng:
                     new_entry['episode_name_eng'] = episode_name_eng
-                entries.append(new_entry)
-
+                    entries.append(new_entry)
         return entries
 
+    def get_torrents(self, task, lostfilm_num, season_num, episode_num):
+        try:
+            response = task.requests.get(V_SEARCH_URL, params={'c': lostfilm_num, 's': season_num, 'e': episode_num})
+        except RequestException as e:
+            log.error('Could not connect to redirect url: %s', e)
+            return {}
+
+        page = get_soup(response.content)
+        try:
+            redirect_url = page.head.meta['content'].split('url=')[1]  # retre.org
+        except:
```

Comment:
This is too broad. Which exceptions are you expecting here? I see potential KeyError and IndexError

Reply:
Broad is ok here, as the idea for all exception catching inside this method is to just skip failed items (returnno terrents) withoud breaking parent cycle and proceed to the next releases. What is missing here is logging the exception info, so I'll add that.